### ISSUE TITLE
Typical `length - 1` can make Rust panic

### ISSUE THREAD

**Issue by Enyium** [rust-lang/rust-clippy#10268]
### What it does

`.len() - 1` may also a problem in other contexts. But I'll keep it confined to what I currently understand about the problem in Rust.

The lint checks for comparison of a value with `something.len() - 1` and suggests to put `+ 1` on the other side of the comparison instead. (Or is there a better solution?)

Possibly also handle bigger values than `1`.

### Lint Name

subtraction_from_len

### Category

correctness, suspicious

### Advantage

`length - 1` is a common programming idiom that you may write in your sleep, not thinking anything bad could come of it. Since `.len()` returns a `usize`, you can't subtract from value `0` without panicking. This lint catches that case that may not be caught otherwise until code is in production.

### Drawbacks

I had cases where I just pushed into a `Vec` and then needed that index. Such cases would be false positives.

### Example

```rust
fn main() {
    let vec = vec![10, 11, 12];
    println!("{}", is_last_index(&vec, 2));

    // Panics: "attempt to subtract with overflow".
    let vec = vec![];
    println!("{}", is_last_index(&vec, 0));
}

fn is_last_index(vec: &Vec<i32>, index: usize) -> bool {
    let last_index = vec.len() - 1;
    index == last_index
}
```

Could be written as:

```rust
fn main() {
    let vec = vec![10, 11, 12];
    println!("{}", is_last_index(&vec, 2));

    let vec = vec![];
    println!("{}", is_last_index(&vec, 0));
}

fn is_last_index(vec: &Vec<i32>, index: usize) -> bool {
    index + 1 == vec.len()
}
```
